jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -481,6 +482,11 @@ private Runnable onFailure(Throwable x, boolean remote)
                 };
 
                 // Serialize all the error actions.
+
+//                Case 2
+                if (invokeOnContentAvailable != null || invokeWriteFailure != null)
+                    invokeOnFailureListeners = null;
```

Comment:
I look at this and it just feels wrong.  To silently throw away listeners in a very specific case does not feel right.
I'm leaning back towards something like my PR... but perhaps with EventListener interfaces?